pkgs/build-support/trivial-builders.nix

```diff
@@ -541,4 +541,31 @@ rec {
       phases = "unpackPhase patchPhase installPhase";
       installPhase = "cp -R ./ $out";
     };
+
+  /* Checks the command output contains the specified version
+   *
+   * Examples:
+   *
+   * passthru.tests.version = testVersion { package = hello; };
+   *
+   * passthru.tests.version = testVersion {
+   *   package = seaweedfs;
+   *   command = "weed version";
+   * };
+   *
+   * passthru.tests.version = testVersion {
+   *   package = key;
+   *   command = "KeY --help";
+   *   # Wrong '2.5' version in the code. Drop on next version.
+   *   version = "2.5";
+   * };
+   */
+  testVersion =
+    { package,
+      command ? "${package.pname or package.name} --version",
```

Comment:
We have a new `meta.mainProgram` on some packages, would that be useful here? It's for cases where the main application's name doesn't match the `pname`, e.g. `7z` in the `p7zip` package.

Reply:
ooh I didn't know about that, that's great, added :+1: